src/daria/analysis/concentrationanalysis.py

```diff
@@ -348,3 +395,383 @@ def _determine_total_volume(self, concentration: daria.Image) -> float:
+            labeled_region = labels_prior == label
+
+            # Determine contour set of labeled region
+            contours, _ = cv2.findContours(
+                skimage.img_as_ubyte(labeled_region),
+                cv2.RETR_TREE,
+                cv2.CHAIN_APPROX_SIMPLE,
+            )
+
+            # For each part of the contour set, check whether the gradient is sufficiently
+            # large at any location
+            accept = False
+            for c in contours:
+
+                # Extract coordinates of contours - have to flip columns, since cv2 provides
+                # reverse matrix indexing, and also 3 components, with the second one
+                # single dimensioned.
+                c = (c[:, 0, 1], c[:, 0, 0])
+
+                # Identify region as marked if gradient sufficiently large
+                if np.max(gradient_modulus[c]) > self.threshold_posterior:
+                    accept = True
+                    break
+            # Collect findings
+            if accept:
+                mask_posterior[labeled_region] = True
+
+        return mask_posterior
+
+    def postprocess_signal(self, signal: np.ndarray) -> np.ndarray:
+        """
+        Postprocessing routine, essentially converting a continuous
+        signal into a binary concentration and thereby segmentation.
+        The post processing consists of presmoothing, thresholding,
+        filling holes, local convex covering, and postsmoothing.
+        Tuning parameters for this routine have to be set in the
+        initialization routine.
+
+        Args:
+            signal (np.ndarray): clean continous signal with values
+                in the range between 0 and 1.
+
+        Returns:
+            np.ndarray: binary concentration
+        """
+        # Determine prior and posterior
+        mask_prior, smooth_signal = self._prior(signal)
+        mask_posterior = self._posterior(smooth_signal, mask_prior)
+
+        # NOTE: Here the overlay process is obsolete, posterior is active.
+        # Yet, it allows to overwrite posterior by inheritance and design
+        # other schemes.
+
+        # Overlay prior and posterior
+        mask = np.zeros(mask_prior.shape, dtype=bool)
+        labels_prior, num_labels_prior = skimage.measure.label(
+            mask_prior, return_num=True
+        )
+
+        for label in range(1, num_labels_prior + 1):
+
+            # Fix one label
+            labeled_region = labels_prior == label
+
+            # Check whether posterior marked in this area
+            if np.any(mask_posterior[labeled_region]):
+                mask[labeled_region] = True
+
+        return mask
+
+    def convert_signal(self, signal: np.ndarray) -> np.ndarray:
+        """
+        Pass the signal, assuming it identifies a binary concentration.
+
+        Args:
+            signal (np.ndarray): signal array.
+
+        Returns:
+            np.ndarray: signal array.
+        """
+        return signal
```

Comment:
Im guess that this method also does something... Perhaps you could tell me what?

Reply:
This is a preparation for allowing to rescale signals depending on the sandlayers. Such a conversion is included in the parent class and scales according to some calibration. The binary concentration class does not contain a calirbation yet and therefore the parent class function is overwritten here.

Reply:
Aah, I see.